--- linux/scripts/package/builddeb	2026-01-26 21:15:45.873754738 +0800
+++ linux/scripts/package/builddeb	2026-01-26 21:18:48.757750200 +0800
@@ -47,9 +47,10 @@
 		cp ${KCONFIG_CONFIG} "${pdir}/usr/share/doc/${pname}/config"
 		gzip "${pdir}/usr/share/doc/${pname}/config"
 	else
-		mkdir -p "${pdir}/boot"
+		mkdir -p "${pdir}/boot/dtbs/qcom"
 		cp System.map "${pdir}/boot/System.map-${KERNELRELEASE}"
 		cp ${KCONFIG_CONFIG} "${pdir}/boot/config-${KERNELRELEASE}"
+		cp "${pdir}/usr/lib/linux-image-${KERNELRELEASE}/qcom/sm8150-"*.dtb "${pdir}/boot/dtbs/qcom/"
 	fi
 
 	# Not all arches have the same installed path in debian
