--- linux/scripts/package/mkdebian	2026-06-03 20:28:03.582474152 +0800
+++ mkdebian.new	2026-06-03 21:50:59.275734619 +0800
@@ -223,17 +223,6 @@
 EOF
 
 if [ "${SRCARCH}" != um ]; then
-cat <<EOF >> debian/control
-
-Package: linux-libc-dev
-Section: devel
-Provides: linux-kernel-headers
-Architecture: $debarch
-Description: Linux support headers for userspace development
- This package provides userspaces headers from the Linux kernel.  These headers
- are used by the installed headers for GNU glibc and other system libraries.
-Multi-Arch: same
-EOF
 
 if is_enabled CONFIG_MODULES; then
 cat <<EOF >> debian/control
@@ -249,18 +238,6 @@
 fi
 fi
 
-if is_enabled CONFIG_DEBUG_INFO; then
-cat <<EOF >> debian/control
-
-Package: linux-image-${KERNELRELEASE}-dbg
-Section: debug
-Architecture: $debarch
-Build-Profiles: <!pkg.${sourcename}.nokerneldbg>
-Description: Linux kernel debugging symbols for ${KERNELRELEASE}
- This package will come in handy if you need to debug the kernel. It provides
- all the necessary debug symbols for the kernel and its modules.
-EOF
-fi
 
 cat <<EOF > debian/rules.vars
 ARCH := ${ARCH}
