diff --git a/arch/arm64/boot/dts/qcom/sm8150-xiaomi-raphael.dts b/arch/arm64/boot/dts/qcom/sm8150-xiaomi-raphael.dts
index 04646b389..735ba2875 100644
--- a/arch/arm64/boot/dts/qcom/sm8150-xiaomi-raphael.dts
+++ b/arch/arm64/boot/dts/qcom/sm8150-xiaomi-raphael.dts
@@ -11,6 +11,7 @@
 #include <dt-bindings/pinctrl/qcom,pmic-gpio.h>
 #include <dt-bindings/regulator/qcom,rpmh-regulator.h>
 #include <dt-bindings/usb/pd.h>
+#include <dt-bindings/sound/qcom,q6afe.h>
 #include "sm8150.dtsi"
 #include "pm8150.dtsi"
 #include "pm8150b.dtsi"
@@ -791,26 +792,6 @@ &qupv3_id_2 {
 	status = "okay";
 };
 
-&remoteproc_adsp {
-	status = "okay";
-	firmware-name = "qcom/sm8150/Xiaomi/raphael/adsp.mbn";
-};
-
-&remoteproc_cdsp {
-	status = "okay";
-	firmware-name = "qcom/sm8150/Xiaomi/raphael/cdsp.mbn";
-};
-
-&remoteproc_mpss {
-	status = "okay";
-	firmware-name = "qcom/sm8150/Xiaomi/raphael/modem.mbn";
-};
-
-&remoteproc_slpi {
-	status = "disabled";
-	firmware-name = "qcom/sm8150/Xiaomi/raphael/slpi.mbn";
-};
-
 &tlmm {
     gpio-line-names = 
         "NFC_ESE_SPI_MISO",     /* GPIO_0 */
@@ -1009,6 +990,29 @@ panel_te_pin: panel-te-state {
         bias-pull-down;
     };
 
+    /* SPKR_I2S_BCK/WS (Nabu: gpio137/138); DOUT on gpio139 (Raphael) */
+    quat_tdm_active: quat-tdm-active-state {
+        pins = "gpio137", "gpio138";
+        function = "qua_mi2s";
+        drive-strength = <8>;
+        bias-disable;
+        output-high;
+    };
+
+    quat_tdm_sd1_active: quat-tdm-sd1-active-state {
+        pins = "gpio139";
+        function = "qua_mi2s";
+        drive-strength = <8>;
+        bias-disable;
+    };
+
+    quat_tdm_sd2_active: quat-tdm-sd2-active-state {
+        pins = "gpio140";
+        function = "qua_mi2s";
+        drive-strength = <8>;
+        bias-disable;
+    };
+
     qup_uart13_default: qup-uart13-default-state {
 	cts-pins {
 		pins = "gpio43";
@@ -1181,3 +1185,93 @@ &wifi {
 
 	status = "okay";
 };
+
+&remoteproc_adsp {
+	status = "okay";
+	firmware-name = "qcom/sm8150/Xiaomi/raphael/adsp.mbn";
+};
+
+&remoteproc_cdsp {
+	status = "okay";
+	firmware-name = "qcom/sm8150/Xiaomi/raphael/cdsp.mbn";
+};
+
+&remoteproc_mpss {
+	status = "okay";
+	firmware-name = "qcom/sm8150/Xiaomi/raphael/modem.mdt";
+};
+
+&remoteproc_slpi {
+	status = "okay";
+	firmware-name = "qcom/sm8150/Xiaomi/raphael/slpi_rp.mdt";
+};
+
+&q6afedai {
+	dai@72 {
+		reg = <QUATERNARY_TDM_RX_0>;
+		qcom,tdm-sync-mode = <1>;
+		qcom,tdm-sync-src = <1>;
+		qcom,tdm-data-out = <0>;
+		qcom,tdm-invert-sync = <1>;
+		qcom,tdm-data-delay = <1>;
+		qcom,tdm-data-align = <0>;
+	};
+};
+
+&q6asmdai {
+	dai@0 {
+		reg = <0>;
+	};
+	dai@1 {
+		reg = <1>;
+	};
+};
+
+&qupv3_id_0 {
+	status = "okay";
+};
+
+&i2c1 {
+	status = "okay";
+	tfa9874: audio-codec@34 {
+		compatible = "nxp,tfa9874";
+		reg = <0x34>;
+		#sound-dai-cells = <0>;
+		vddd-supply = <&vreg_l7a_1p8>;
+		vdda-supply = <&vreg_l10c_3p3>;
+		reset-gpios = <&tlmm 59 GPIO_ACTIVE_HIGH>;
+		interrupt-parent = <&tlmm>;
+		interrupts = <141 IRQ_TYPE_LEVEL_LOW>;
+		status = "okay";
+	};
+};
+
+&sound {
+	compatible = "qcom,sm8150-sndcard";
+	model = "xiaomi-raphael";
+	pinctrl-0 = <&quat_tdm_active &quat_tdm_sd1_active &quat_tdm_sd2_active>;
+	pinctrl-names = "default";
+	clocks = <&q6afecc 102 0>,
+		 <&q6afecc 103 0>;
+	clock-names = "lpass_macro", "lpass_dcodec";
+	audio-routing = "HiFi Playback", "Quaternary TDM0 Playback", "Speaker", "HiFi Playback";
+	mm1-dai-link {
+		link-name = "MultiMedia1";
+		cpu {
+			sound-dai = <&q6asmdai 0>;
+		};
+	};
+	primary-tdm-dai-link {
+		no-pcm;
+		link-name = "Speaker Playback";
+		cpu {
+			sound-dai = <&q6afedai QUATERNARY_TDM_RX_0>;
+		};
+		platform {
+			sound-dai = <&q6routing>;
+		};
+		codec {
+			sound-dai = <&tfa9874>;
+		};
+	};
+};
