diff --git a/arch/arm64/boot/dts/qcom/sm8150.dtsi b/arch/arm64/boot/dts/qcom/sm8150.dtsi
index 079e2f0a6..131a0d8b5 100644
--- a/arch/arm64/boot/dts/qcom/sm8150.dtsi
+++ b/arch/arm64/boot/dts/qcom/sm8150.dtsi
@@ -794,6 +794,12 @@ qseecom_mem: memory@9e400000 {
 			reg = <0x0 0x9e400000 0x0 0x1400000>;
 			no-map;
 		};
+
+		fastrpc_mem: memory@ae800000 {
+			reg = <0x0 0xae800000 0x0 0x2000000>;
+			no-map;
+		};
+
 	};
 
 	smem {
@@ -2279,6 +2285,8 @@ fastrpc {
 					qcom,non-secure-domain;
 					#address-cells = <1>;
 					#size-cells = <0>;
+					memory-region = <&fastrpc_mem>;
+					qcom,vmids = <QCOM_SCM_VMID_HLOS QCOM_SCM_VMID_MSS_MSA QCOM_SCM_VMID_SSC_Q6 QCOM_SCM_VMID_ADSP_Q6>;
 
 					compute-cb@1 {
 						compatible = "qcom,fastrpc-compute-cb";
@@ -3455,6 +3463,7 @@ fastrpc {
 					qcom,non-secure-domain;
 					#address-cells = <1>;
 					#size-cells = <0>;
+					
 
 					compute-cb@1 {
 						compatible = "qcom,fastrpc-compute-cb";
@@ -4499,6 +4508,7 @@ fastrpc {
 					qcom,non-secure-domain;
 					#address-cells = <1>;
 					#size-cells = <0>;
+					
 
 					compute-cb@3 {
 						compatible = "qcom,fastrpc-compute-cb";
