diff --git a/sound/soc/codecs/tfa9872.c b/sound/soc/codecs/tfa9872.c
index 26bed4305..9424a7460 100644
--- a/sound/soc/codecs/tfa9872.c
+++ b/sound/soc/codecs/tfa9872.c
@@ -6,6 +6,7 @@
  */
 
 #include <linux/bitfield.h>
+#include <linux/gpio/consumer.h>
 #include <linux/i2c.h>
 #include <linux/module.h>
 #include <linux/regmap.h>
@@ -219,9 +220,21 @@ static int tfa987x_i2c_probe(struct i2c_client *i2c)
 {
 	struct device *dev = &i2c->dev;
 	struct regmap *rmap;
+	struct gpio_desc *reset_gpiod;
 	unsigned int rev;
 	int ret;
 
+	reset_gpiod = devm_gpiod_get_optional(dev, "reset", GPIOD_OUT_LOW);
+	if (IS_ERR(reset_gpiod))
+		return PTR_ERR(reset_gpiod);
+
+	if (reset_gpiod) {
+		gpiod_set_value_cansleep(reset_gpiod, 1);
+		usleep_range(1000, 2000);
+		gpiod_set_value_cansleep(reset_gpiod, 0);
+		usleep_range(1000, 2000);
+	}
+
 	rmap = devm_regmap_init_i2c(i2c, &tfa987x_regmap_config);
 	if (IS_ERR(rmap))
 		return PTR_ERR(rmap);
@@ -263,7 +276,7 @@ static int tfa987x_i2c_probe(struct i2c_client *i2c)
 	regmap_update_bits(rmap, TFA987X_TDM_CFG1,
 				 TFA987X_TDM_CFG1_NSLOTS_MSK |
 				 TFA987X_TDM_CFG1_SLOTBITS_MSK,
-				 FIELD_PREP(TFA987X_TDM_CFG1_NSLOTS_MSK, 1) |
+				 FIELD_PREP(TFA987X_TDM_CFG1_NSLOTS_MSK, 8) |
 				 FIELD_PREP(TFA987X_TDM_CFG1_SLOTBITS_MSK, 15));
 	regmap_update_bits(rmap, TFA987X_TDM_CFG2,
 				 TFA987X_TDM_CFG2_SWIDTH_MSK,
